stdlib/cache/src/main/ballerina/src/cache/cache.bal

```diff
@@ -151,15 +151,17 @@ public type Cache object {
 
     # Evicts the cache when the cache is full.
     function evict() {
-        int maxCapacity = self.capacity;
-        float ef = self.evictionFactor;
-        int numberOfKeysToEvict = <int>(maxCapacity * ef);
-        // Get the above number of least recently used cache entry keys from the cache
-        string[] cacheKeys = self.getLRUCacheKeys(numberOfKeysToEvict);
-        // Iterate through the map and remove entries.
-        foreach var c in cacheKeys {
-            // These cache values are ignored. So it is not needed to check the return value for the remove function.
-            var tempVar = self.entries.remove(c);
+        lock {
```

Comment:
Can we add new entries to cache while evict function execute or lock prevent the access to the map?

Reply:
It prevents access to the map.

Reply:
This will drastically affect the performance of caching. Can't we only delay the start time and refain from using locking?

Reply:
We can, but then again, earlier the map access was anyway concurrent safe. Now that it's not, it's advised to handle it by the user.

Reply:
Anyway, revert locks on 2a25a0f.

Reply:
We can alter to use this cache implementation without relying on tasks. A community user already did this and we might able to add that change. So for the moment let remove the locking